src/plugins/kibana_react/public/use_execution_context/use_execution_context.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { KibanaExecutionContext, CoreStart } from 'kibana/public';
+import useDeepCompareEffect from 'react-use/lib/useDeepCompareEffect';
+
+/**
+ * Set and clean up application level execution context
+ * @param executionContext
+ * @param context
+ */
+export function useExecutionContext(
+  executionContext: CoreStart['executionContext'],
+  context: KibanaExecutionContext
+) {
+  useDeepCompareEffect(() => {
```

Comment:
Do we need this `deep` compare? doesn't service logic ignores "unnecessary" updates?

Reply:
Apps often re-render the main component, while the context does not change. I don't want the `context$` to do work unnecessarily. No?